osu.Game/Skinning/SkinProvidingContainer.cs

```diff
@@ -38,89 +41,112 @@ public class SkinProvidingContainer : Container, ISkinSource
         protected virtual bool AllowColourLookup => true;
 
         public SkinProvidingContainer(ISkin skin)
+            : this()
         {
-            this.skin = skin;
+            SkinLayers.Add(skin);
+        }
 
+        protected SkinProvidingContainer()
+        {
```

Comment:
Pretty flaky API here. If an implementation happened to call the `ctor` with a skin parameter then `SkinLayers.Clear` as you're doing, that skin would be lost.

If we're actually using this container with a single skin (are we?) maybe it's best to have that as its own implementation and make this class abstract (and never add to the "layers" in its own `ctor`).

Reply:
`SkinProvidingContainer` is 99.9% of the time used with a single skin in `ctor`, I was very unsure about making `SkinProvidingContainer` abstract because of the naming, can you think of a suitable one?

I've went with a public ctor that adds `SkinLayers` for the public instantiable side of `SkinProvidingContainer`, if an implementation accesses `SkinLayers` then it is on their responsibility, or maybe we could just disable mutating `SkinLayers` in some way on this ctor to avoid such mistake?

Reply:
Not sure. I'd probably be okay with just xmldoc since it's a `protected ctor` but others may have different opinions.